app/_plugins/generators/latest_version_generator.rb

```diff
@@ -10,54 +10,64 @@ class Page
-      products_with_latest = %w[gateway mesh KIC deck]
 
-      @page_index = site.config['defaults'].to_h do |s|
-        next [s['scope']['path'], s['values']['version-index']]
-      end
+    PRODUCTS_WITH_LATEST = {
+      'gateway' => 'gateway',
+      'mesh' => 'mesh',
+      'deck' => 'deck',
+      'kubernetes-ingress-controller' => 'KIC'
+    }.freeze
+
+    def generate(site)
+      @site = site
 
-      # Load config file
       site.pages.each do |page|
-        page_path = remove_generated_prefix(page)
-        parts = Pathname(page_path).each_filename.to_a
-
-        products_with_latest.each do |product|
-          # Reset values for every new page
-          generate_latest = false
-          release_path = nil
-
-          product_name = product.downcase
-          # Special case KIC
-          product_name = 'kubernetes-ingress-controller' if product_name == 'kic'
-
-          # Latest version
-          if parts[0] == product_name && parts[1] == site.data["kong_latest_#{product}"]['release']
-            generate_latest = true
-            release_path = parts[1]
-          end
-
-          next unless generate_latest && !page.data['is_latest']
-          # If it has a permalink it's _probably_ an index page e.g. /gateway/
-          # so we should not generate a /latest/ URL as it's already evergreen
-          next if page.data['permalink']
-
-          # Otherwise, let's generate a /latest/ URL too
-          page = DuplicatePage.new(
-            site,
-            site.source,
-            page.url.gsub(release_path, 'latest'),
-            page.content,
-            page.data,
-            @page_index["#{product_name}/#{release_path}/"],
-            page_path,
-            page.relative_path
-          )
-          site.pages << page
-        end
+        # If it has a permalink it's _probably_ an index page e.g. /gateway/
+        # so we should not generate a /latest/ URL as it's already evergreen
+        next if page.data['is_latest'] || page.data['permalink']
+
+        next unless generate_latest?(page)
+
+        # Otherwise, let's generate a /latest/ URL too
+        site.pages << build_duplicate_page(page)
       end
     end
 
+    private
+
+    def build_duplicate_page(page)
+      product_name, release_path = page_path(page).first(2)
+
+      DuplicatePage.new(
+        @site,
+        page.url.gsub(release_path, 'latest'),
+        page,
+        page_index["#{product_name}/#{release_path}/"],
+        remove_generated_prefix(page)
+      )
+    end
+
+    def generate_latest?(page)
+      product_name_from_path, release_path = page_path(page).first(2)
+      product_name, product = PRODUCTS_WITH_LATEST.detect do |k, _|
+        k == product_name_from_path
+      end
+
+      product_name &&
+        release_path == @site.data["kong_latest_#{product}"]['release']
+    end
+
+    def page_index
+      # XXX: this isn't being set on any of the pages
```

Comment:
This is a legacy thing. We used to set `version-index` in `jekyll.yml` but I replaced it with `page.versions.ce` and `page.versions.ee`

Reply:
Looks like it is still used in [older pages](https://github.com/Kong/docs.konghq.com/search?q=version-index) that are not single-sourced. It looks like it's also used by some `_includes` so I'm keeping this and get back to it in a follow-up ticket